src/index.js

```diff
@@ -93,20 +104,68 @@ export default class Scrl {
             // this.triggerEvent('scrollDone')
         }
     }
-
-    _update = () => {
+    
+    /**
+     * Updates and returns the distance the scroll has still to go
+     * @returns float
+     */
+    _getUpdatedDistance = () => {
         const distance = this._targetPositionYWithOffset - this._positionY;
         const attraction = distance * this.options.acceleration;
-
+        
         this._velocityY += attraction;
-
+        
         this._velocityY *= this.options.friction;
         this._positionY += this._velocityY;
-
+        
         return Math.abs(distance);
     }
-
+    
+    /**
+     * Apply the current scroll position to either the window or the HTMLElement
+     */
     _render = () => {
-        window.scrollTo(0, this._positionY);
+        this.options.element.scrollTo(0, this._positionY);
+    }
+    
+    /**
+    * Checks if an the given options.element is equal to window
+    * @see https://stackoverflow.com/a/3099796/586823
+    * @returns boolean
+    */
+    _elementIsWindow() {
+        return this.options.element.self === window;
+    }
+
+    /**
+     * Get the target scroll position for a given target element
+     * @param {window|HTMLElement} targetElement 
+     * @returns float
+     */
+    _getTargetPositionFromElement(targetElement) {
+        if( this._elementIsWindow() ) {
+            return Math.round(targetElement.getBoundingClientRect().top + window.pageYOffset)
+        }
+        return targetElement.getBoundingClientRect().top - this.options.element.getBoundingClientRect().top + this.options.element.scrollTop;
+    }
+    
+    /**
+     * Get the maximum scroll position
+     * @returns float
+     */
+    _getMaximumPossibleScrollPosition() {
+        if( this._elementIsWindow() ) {
+            return document.documentElement.scrollHeight - window.innerHeight;
+        }
+        return this.options.element.scrollHeight - this.options.element.offsetHeight;
+    }
+    
+    /**
+    * Returns either the window's or an element's current scroll position
+    * @returns float
+    */
+    _getScrollPosition() {
+        return this._elementIsWindow() ? window.pageYOffset : this.options.element.scrollTop;
     }
+    
```

Comment:
All of these functions are happening many times a second, each time going through the check of window/element several times, which is not great performance-wise. I think the check should be done once on initialization, and the correct methods would be set to be used at that time, later avoiding the check altogether. WDYT?

Reply:
Actually, all of these functions are only being called once each time `scrollTo()` is being invoked, so we shouldn't run into performance problems here. I opted for using the function `_elementIsWindow` instead of a class property, because I wanted the script to reliably work even if `options.element` was being changed after the class instantiation. 

So this would still work as expected:

```js
const scrl = new Scrl({
  element: window
});
scrl.options.element = document.querySelector('#my-overflowing-div');
```

Reply:
...changed the option to `scrollTarget`